autogluon/task/text_classification/dataset.py

```diff
@@ -119,28 +122,167 @@ def get_dataset(self, segment='train'):
+    doublequote : bool, default ``True``
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    escapechar : str (length 1), optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    comment : str, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    encoding : str, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    dialect : str or csv.Dialect, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    error_bad_lines : bool, default True
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    warn_bad_lines : bool, default True
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    delim_whitespace : bool, default False
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    low_memory : bool, default True
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    memory_map : bool, default False
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    float_precision : str, optional
+        See <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.read_csv.html#pandas.read_csv`_ .
+    """
+    def __init__(self, *args, **kwargs):
+        self._read(**kwargs)
+        self.is_pair = False
+        self.class_labels = list(set([sample[1] for sample in self.dataset]))
+        self.metric = Accuracy()
+        super(CustomTSVClassificationTask, self).__init__(self.class_labels, self.metric, self.is_pair)
+
+    def get_dataset(self, segment='train'):
+        """Get the corresponding dataset for the task.
+
+        Parameters
+        ----------
+        segment : str, default 'train'
+            Dataset segments.
+
+        Returns
+        -------
+        The dataset of target segment.
+        """
+        if segment == 'train':
+            return SampledDataset(self.dataset, self.train_sampler)
+        elif segment == 'dev':
+            return SampledDataset(self.dataset, self.dev_sampler)
+        else:
+            raise NotImplementedError
+
+    def _read(self, **kwargs):
+        path = kwargs.get('filepath', None)
+        kwargs['filepath_or_buffer'] = path
+        kwargs.pop("filepath")
+        dataset_df = pd.read_csv(**kwargs)
+        dataset_df_lst = dataset_df.values.tolist()
+        self.dataset = gluon.data.SimpleDataset(dataset_df_lst)
+        split_ratio = kwargs.get('split_ratio', None) if 'split_ratio' in kwargs else 0.8
```

Comment:
Just have it be a default value in __init__, this is unnecessary

Reply:
I think having a default value means an additional required argument, which is actually unnecessary.

Reply:
A default value by definition means that it is not a required argument, because the user doesn't have to specify it (default is used if not specified).

Reply:
What I mean is the following: def(split_ratio=0.8, **kwargs) comparing to def(**kwargs), there is an extra argument in the function definition.

Reply:
Yes, I think it is good to have instead of just only **kwargs. We should honestly minimize the amount of parameters that have to be specified through kwargs. For example, Tabular only has very niche parameters as kwargs.

Reply:
Yes, this is a design preference. I would rather provide a simplified interface with a doc string explaining what's available. Agree on we should specify which keyword arguments are available to the customers.